### ISSUE TITLE
Calling an artifact with two sources fails.

### ISSUE THREAD

**Issue by scudette** [Velocidex/velociraptor#1235]
With an artifact like this

```yaml
name: MultiSourceSerialMode
sources:
  - name: Source1
    precondition: SELECT * FROM info() WHERE OS = "linux"
    query: |
      SELECT * from info()
  - name: Source2
    precondition: SELECT * FROM info() WHERE OS = "windows"
    query: |
      SELECT * FROM info()
```

Calling 
```sql
select * from Artifact.MultiSourceSerialMode(preconditions=TRUE)
```

Returns an error - it should just run both sources one after the other.

**Comment by predictiple** [comment]
![Screenshot from 2021-09-05 16-00-46](https://user-images.githubusercontent.com/9843422/132130695-1b68477e-b0d5-49d9-bd0b-6d6ca3d45a79.png)

**Comment by predictiple** [comment]
It's probably worth noting that with `preconditions=false` (the default for the Artifact() plugin) the results from multiple sources are chained, which apparently doesn't cause any issue. It would be consistent behaviour if `preconditions=true` did the same.

In the case where it is desirable to have only 1 source return a result then a switch() statement could be used to call each Artifact source.

**Comment by predictiple** [comment]
There's something not quite right still.

If `select * from Artifact.MultiSourceSerialMode(preconditions=TRUE)` is run _in a notebook_ then the preconditions work. With the above MultiSourceSerialMode artifact that produces 1 result row, as expected.

If `select * from Artifact.MultiSourceSerialMode(preconditions=FALSE)` is run _in a notebook_ then the preconditions are ignores. With the above MultiSourceSerialMode artifact that produces 2 result rows, as expected.

If the Artifact plugin is called from another artifact, e.g. the example below, then preconditions are ignored. That is `preconditions=TRUE` and `preconditions=FALSE` both produce the same 2-row result.

```yaml
name: Custom.Artifact.TestPreconditions
description: |

type: CLIENT

sources:
  - query: |
      select * from Artifact.MultiSourceSerialMode(preconditions=TRUE)
```
Tested used today's CI build for both server and client:
version: 0.6.1
commit: ccfad72
build_time: "2021-09-07T08:45:13Z"

**Comment by predictiple** [comment]
Even if a specific source in the child-artifact is specified in the parent artifact 
`select * from Artifact.MultiSourceSerialMode(preconditions=TRUE, source="Source2")`
the preconditions are ignored.

**Comment by predictiple** [comment]
I've now also tested with:
version: 0.6.0
commit: 5957468b
build_time: "2021-06-25T02:05:13+10:00"

In this old version we had to specify a source when using `preconditions=true` but again there is the same difference in behaviour when the Artifact plugin is called from a notebook vs. being called from an artifact. When called from an artifact the preconditions are ignored regardless of the flag.

So the problem is apparently not directly related to the #1237 fix as it was pre-existing. The behaviour of the Artifacts plugin needs to be the same when called from artifacts as it is now when called from notebooks.
